Publish TWIM (#2784)
matrix-org/matrix.org: content/blog/2025/06/2025-06-13-twim.md

+++

> date = "2025-06-13"

Reviewer: need to sort after the premium accounts
```suggestion
date = "2025-06-13T16:00:00Z"
```

title = "This Week in Matrix 2025-06-13"
path = "/blog/2025/06/13/this-week-in-matrix-2025-06-13"

[taxonomies]
author = ["Thib (m.org)"]
category = ["This Week in Matrix"]
+++

## Matrix Live

> {{youtube_player(video_id="7v1dYUk_Bbk ")}}

Reviewer: ```suggestion
{{ youtube_player(video_id="7v1dYUk_Bbk")}}
```

## Dept of *Status of Matrix* 🌡️

[Thib](https://matrix.to/#/@thibaultmartin:matrix.org) reports

> > Earlier Today, the Matrix.org Foundation introduced Premium Accounts to fund the matrix.org homeserver. This requires little to no change for client developers.

Reviewer: ```suggestion
> Earlier Today, the Matrix.org Foundation announced its plans to introduce Premium Accounts to fund the matrix.org homeserver. This requires little to no change for client developers.
```

> 

> > The full announcement is at https://matrix.org/blog/2025/06/funding-homeserver-premium/

Reviewer: ```suggestion
> The full announcement is at <https://matrix.org/blog/2025/06/funding-homeserver-premium/>
```

> 

Reviewer: ```suggestion

<!-- more -->

```

> 
> On a final note, we're still hard at work building out the new interactive commands / element timeline rendering features for Hookshot, which is taking a bit more time due to the number of PRs that need to land (at the last count, 5 are involved in the full feature set). This is happening now though, and I hope that next week we will be bringing you a release that covers it all. Until next time!

## Dept of Clients 📱

### Extera Next

[OfficialDakari](https://matrix.to/#/@officialdakari:extera.xyz) reports

> Hello! During this week, a few changes happened in Extera Next:
> * Migration to Vodozemac. Extera Next now uses Vodozemac too, as FluffyChat (i totally didn't steal any code)
> * Fixed plaintext password dialog when deactivating account or removing a device
> * Fixed terrible design of lock screen. Removed app banner and changed input placeholder to "PIN code" instead of 4 asterisks 
> * Removed send image/video buttons. Send file can do the same, why to have them?
> 

> > Download APK: https://extera.xyz/next/next.apk?_cacheBuster
> > Git: https://git.extera.xyz/OfficialDakari/ExteraNext

Reviewer: ```suggestion
> Download APK: <https://extera.xyz/next/next.apk?_cacheBuster>
> Git: <https://git.extera.xyz/OfficialDakari/ExteraNext>
```

> * Threads are now available as an experimental developer option! They are still WIP and may lead to unexpected behaviour, but if you want you can try them out.
> * some accessibility improvements were made across the app and many more are to come!
> * Fixed a bug that was crashing the app on the new iOS 26 beta versions

## Dept of Non Chat Clients 🎛️

[andybalaam](https://matrix.to/#/@andybalaam:matrix.org) says

> Announcing `mx`: an experimental command line tool for messing with your Matrix account.
> 
> This is a tool that is intended to run as part of scripts and automation, as opposed to being interactive. It's written in Rust.
> 
> Example:
> 
> ```
> $ mx login https://chat.example.com myusername
> ...
> $ mx recovery status
> ...
>     master_key=rPnI6NjlQcJRuEEaTwpAOv41k/dkh8ckgdTIJXNE+xU
>     user_signing_key=jsdfFsdsd04fks93r/3r2fufwojiwef9FEJfeijdeDD
>     self_signing_key=iFeZWnl/XscWzztml2xh8RlkqaLKa78ekK4ncJFnSy8
> ```
> 
> It's mostly good for encryption-adjacent things for now, but I vaguely plan to allow it to send messages and similar things in future.
> 

> > Find out more and contribute at https://codeberg.org/andybalaam/mx

Reviewer: ```suggestion
> Find out more and contribute at <https://codeberg.org/andybalaam/mx>
```

> 
> #### Crypto
> 
> * [#5219](https://github.com/matrix-org/matrix-rust-sdk/pull/5219) If the SDK believes that the sender of a room event has been spoofed, the SDK will use a dedicated verification level, possibly allowing us to treat such events more carefully. This new verification level is `MismatchedSender`.
> 
> #### General
> 
> * [#5205](https://github.com/matrix-org/matrix-rust-sdk/pull/5205) We are reverting our experiment around `busy_timeout` with SQLite to compare the data and see if it's helping fixing some lock issues.
> * [#5099](https://github.com/matrix-org/matrix-rust-sdk/pull/5099) When using `Client::add_event_handler`, it is now possible to define a callback with an argument of kind `Option<EncryptionInfo>`, which is particularly useful and requested by the community frequently!

### Matrix Dart SDK ([website](https://github.com/famedly/matrix-dart-sdk))

Matrix SDK written in pure Dart.

[td](https://matrix.to/#/@td:technodisaster.com) reports

> #### Released v1.0.0 this week
> 

> > * This major new release now move away from libolm to vodozemac. Fluffychat now has a call for testing open here https://github.com/krille-chan/fluffychat/issues/1957
> > * v1.14 spec endpoints support
> > * A bunch of other breaking changes since the last announcement, https://github.com/famedly/matrix-dart-sdk/releases

Reviewer: ```suggestion
> * This major new release now move away from libolm to vodozemac. Fluffychat now has a call for testing open here <https://github.com/krille-chan/fluffychat/issues/1957>
> * v1.14 spec endpoints support
> * A bunch of other breaking changes since the last announcement, <https://github.com/famedly/matrix-dart-sdk/releases>
```
